sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RemoveNoopOperatorsSuite.scala

```diff
@@ -44,6 +45,22 @@ class RemoveNoopOperatorsSuite extends PlanTest {
     comparePlans(optimized, testRelation)
   }
 
+  test("Do not remove projections with non-attribute expressions that reuse expr ids") {
```

Comment:
It seems that all newly added UTs fails with the following exception.
```
org.apache.spark.sql.catalyst.errors.package$TreeNodeException:
The structural integrity of the input plan is broken in org.apache.spark.sql.catalyst.analysis.SimpleAnalyzer., tree:
```

Reply:
Yeah, let me take another look. Internally I am developing against a slightly older version of the codebase, so there might be some differences.